transformers/modeling_utils.py

```diff
@@ -450,6 +450,26 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
             for old_key, new_key in zip(old_keys, new_keys):
                 state_dict[new_key] = state_dict.pop(old_key)
 
+            # initialize new position embeddings
+            _k = 'bert.embeddings.position_embeddings.weight'
+            if _k in state_dict and config.max_position_embeddings != state_dict[_k].shape[0]:
+                logger.info("config.max_position_embeddings != state_dict[bert.embeddings.position_embeddings.weight] ({0} - {1})".format(
```

Comment:
ok this part scares me a little because it can effect other models, so need to be more cautious.
Would be much better to do is resize after we load the model with a `resize_position_embeddings` method, if possible. 

If that's not possible, I would rather only support the default max_position_embeddings than do this in such a central place.